tstest/integration/integration_test.go

```diff
@@ -311,6 +311,96 @@ func TestAddPingRequest(t *testing.T) {
 	t.Error("all ping attempts failed")
 }
 
+func TestTaildrop(t *testing.T) {
+	t.Parallel()
+	bins := BuildTestBinaries(t)
+
+	env := newTestEnv(t, bins, configureControl(func(control *testcontrol.Server) {
+		control.AllNodesSameUser = true
+	}))
+	defer env.Close()
+
+	n1 := newTestNode(t, env)
+	n1SocksAddrCh := n1.socks5AddrChan()
+	d1 := n1.StartDaemon(t)
+	defer d1.Kill()
+
+	n2 := newTestNode(t, env)
+	n2SocksAddrCh := n2.socks5AddrChan()
+	d2 := n2.StartDaemon(t)
+	defer d2.Kill()
+
+	n1Socks := n1.AwaitSocksAddr(t, n1SocksAddrCh)
+	n2Socks := n1.AwaitSocksAddr(t, n2SocksAddrCh)
+	t.Logf("node1 SOCKS5 addr: %v", n1Socks)
+	t.Logf("node2 SOCKS5 addr: %v", n2Socks)
+
+	for _, n := range env.Control.AllNodes() {
+		n.Capabilities = append(n.Capabilities, tailcfg.CapabilityFileSharing)
+	}
+
+	n1.AwaitListening(t)
+	n2.AwaitListening(t)
+	n1.MustUp()
+	n2.MustUp()
+	n1.AwaitRunning(t)
+	n2.AwaitRunning(t)
+
+	target := n2.AwaitIP(t)
+
+	tempDir := t.TempDir()
+	tmpFile, err := ioutil.TempFile(tempDir, "taildrop")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(tmpFile.Name())
+	contents := []byte("Taildrop drop bop 💧 ??%@#@123˙©∆∆˚ 水平線")
+	if _, err = tmpFile.Write(contents); err != nil {
+		t.Errorf("Failed to write to file: %v", err)
+	}
+
+	getCmd := n2.Tailscale("file", "get", "-wait", tempDir)
+	if err := getCmd.Start(); err != nil {
+		t.Fatal(err)
+	}
+	targetsOutput, err := n1.Tailscale("file", "cp", "-targets").CombinedOutput()
+	if err != nil {
+		t.Fatal(string(targetsOutput), err)
+	}
+	if !bytes.Contains(targetsOutput, []byte(target.String())) {
+		t.Errorf("Missing target from cp -targets, want: %v, in: %v", target, targetsOutput)
+	}
+
+	// Need ALL_PROXY=SOCKS because peerapi was not configured to work with userspace-networking
+	// yet.
+	cpCmd := n1.Tailscale("ALL_PROXY=SOCKS", "file", "cp", tmpFile.Name(), fmt.Sprintf("%s:", target))
```

Comment:
what is this ALL_PROXY=SOCKS doing here? it's being passed as an argument? to the cmd/tailscale command? I'm missing something. How does this work?

Reply:
hmmmmm I have no idea how it works exactly, but by setting this it makes some `os.Setenv` cause it to use SOCKS5 lol
it's not a direct argument though, just an env var

Reply:
oh wait this is just broken never mind